dashboard/app/controllers/courses_controller.rb

```diff
@@ -6,8 +6,8 @@ class CoursesController < ApplicationController
   def index
     respond_to do |format|
       format.html do
-        @is_teacher = !!(current_user && current_user.teacher?)
-        @is_english = request.language == 'en'
+        @is_teacher = (current_user && current_user.teacher?) || (!current_user && params[:view] == 'teacher')
+        @is_english = request.language == 'en' && params[:lang] != 'nonen'
```

Comment:
Ah, I put that in to help debug the page more quickly, but it's a bit confusing so will remove.